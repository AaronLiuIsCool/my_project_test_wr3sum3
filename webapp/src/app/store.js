import { configureStore } from '@reduxjs/toolkit';
import resumeReducer from 'features/SmartResume/slicers/resumeSlicer';
import counterReducer from 'features/CounterExample/counterSlice.js';

export default configureStore({
  reducer: {
    counter: counterReducer,
    resume: resumeReducer
  },
});
