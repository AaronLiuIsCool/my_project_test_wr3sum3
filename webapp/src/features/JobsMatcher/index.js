import React, { Suspense } from 'react';

import LoadFallbackComponent from 'components/LoadFallbackComponent';
import './styles/index.scss';

const JobMatcher = React.lazy(() => import('./components/JobsMatcher'));

const LazyLoadJobMatcher = () => (
  <Suspense fallback={<LoadFallbackComponent />}>
    <JobMatcher />
  </Suspense>
);

export default LazyLoadJobMatcher;
