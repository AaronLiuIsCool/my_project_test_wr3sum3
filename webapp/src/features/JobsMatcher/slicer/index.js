import { createSlice } from '@reduxjs/toolkit';

const jobsMatcherSlicers = createSlice({
  name: 'jobs',
  initialState: {
    bookmarked: []
  },
  reducers: {
    addBookmark: (state, action) => {
      const jobUUID = action.payload;
      if (!jobUUID) {
        return;
      }
      state.bookmarked.indexOf(jobUUID) === -1 && state.bookmarked.push(jobUUID);
    },
    removeBookmark: (state, action) => {
      const forRemoval = action.payload;
      if (!forRemoval) {
        return;
      }
      state.bookmarked = state.bookmarked.filter(jobUUID => jobUUID !== forRemoval);
    },
    setBookmarkedList: (state, action) => {
      state.bookmarked = action?.payload || [];
    }
  },
});

export const { addBookmark, removeBookmark, setBookmarkedList } = jobsMatcherSlicers.actions;

export const selectBookmarkedJobs = ({jobs}) => jobs.bookmarked;

export default jobsMatcherSlicers.reducer;
