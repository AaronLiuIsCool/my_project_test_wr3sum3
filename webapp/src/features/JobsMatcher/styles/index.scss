.job-matcher {
  margin: auto;
  width: 1000px;

  // TODO: refactor below
  .form_item {
    label {
      color: #bfc4d3;
    }

    input {
      height: 41px;
      background-color: #f2f4f9;
      color: #333333;
      border-radius: 8px;
      border: none;
      
      ::placeholder, :-ms-input-placeholder, ::-ms-input-placeholder  {
        color: #d5d9e6;
      }
    }
  }

  ul.pagination {
    .page-link {
      width: 40px;
      height: 40px;
      border-radius: 8px;
      border-color: #f2f4f9;
      background-color: #f2f4f9;
      margin-right: 16px;
      color: #bfc4d3;
      text-align: center;
    }

    .active {
      .page-link {
        border-color: #bfc4d3;
        background-color: #bfc4d3;
        color: #ffffff;
      }
    }
  }

  circle {
    transition: stroke-dashoffset 0.35s;
    transform: rotate(-90deg);
    transform-origin: 50% 50%;
  }

  .score-ring {
    margin-top: 4px;
    position: absolute;
  }
}