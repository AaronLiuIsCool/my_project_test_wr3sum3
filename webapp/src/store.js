import { configureStore } from '@reduxjs/toolkit';

import appReducer from 'features/App/slicer';
import resumeReducer from 'features/SmartResume/slicer';

export default configureStore({
  reducer: {
    app: appReducer,
    resume: resumeReducer
  },
});
