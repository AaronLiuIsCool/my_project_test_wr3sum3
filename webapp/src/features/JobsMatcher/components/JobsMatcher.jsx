import React, { useState } from 'react';
import { useSelector } from 'react-redux';

import JobsService from 'shell/services/JobsServices';
import { I8nContext } from 'shell/i18n';
import { selectLanguage } from 'features/App/slicer';

import SearchHeader from './SearchHeader';
import Jobs from './Jobs';

import zh from '../i18n/zh.json';
import en from '../i18n/en.json';

const jobServices = new JobsService();

const JobMatcher = () => {
    const language = useSelector(selectLanguage);
    const messages = language === 'zh' ? zh : en;

    const [searchResults, setSearchResults] = useState({});
    const [resultsPageNumber, setResultsPageNumber] = useState(0);
    const [query, setQuery] = useState();
    const [country, setCountry] = useState();
    const [city, setCity] = useState();

    const search = async(query, country, city, pageNumber) => {
        const data = await jobServices.getJobs(query, country, city, pageNumber);
        setSearchResults(data);
    }

    const handleSearch = (query, country, city) => {
        setSearchResults({});
        setResultsPageNumber(0);
        setQuery(query);
        setCountry(country);
        setCity(city);

        search(query, country, city, 0);
    };

    const handlePageChange = (pageNumber) => {
        setResultsPageNumber(pageNumber);
        search(query, country, city, pageNumber);
    }

    return (
        <I8nContext.Provider value={messages}>
            <div className="features job-matcher">
                <SearchHeader onSearch={handleSearch} />
                <Jobs data={searchResults} pageNumber={resultsPageNumber}
                    onPageChange={handlePageChange} />
            </div>
        </I8nContext.Provider>
    );
};

export default JobMatcher;