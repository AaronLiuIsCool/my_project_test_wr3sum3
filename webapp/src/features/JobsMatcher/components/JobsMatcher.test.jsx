import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import store from 'store';

import JobsMatcher from './JobsMatcher';


test('renders JobsMatcher', () => {
    const jobsMatcher = renderer.create(
        <Provider store={store}>
            <JobsMatcher />
        </Provider>
    ).toJSON();

    expect(jobsMatcher).toMatchSnapshot();
});
