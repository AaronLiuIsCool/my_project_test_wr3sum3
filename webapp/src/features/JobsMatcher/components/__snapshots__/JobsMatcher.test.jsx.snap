// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders JobsMatcher 1`] = `
<div
  className="features job-matcher"
>
  <form
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-6"
      >
        <div
          className="form_item form-group"
        >
          <label
            className="form-label"
            htmlFor="searchQuery"
          >
            工作岗位搜索
          </label>
          <div
            className="input-group"
          >
            <input
              className="form-control"
              id="searchQuery"
              input="text"
              onChange={[Function]}
              placeholder="输入职位或公司"
              value=""
            />
            <div
              className="input-group-append"
            >
              <button
                className="search-button btn btn-primary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                搜索
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-lg-2"
      >
        <div
          className="form_item form-group"
        >
          <label
            className="form-label"
            htmlFor="searchCountry"
          >
            国家
          </label>
          <input
            className="form-control"
            id="searchCountry"
            onChange={[Function]}
            placeholder="国家"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="col-lg-2"
      >
        <div
          className="form_item form-group"
        >
          <label
            className="form-label"
            htmlFor="searchCity"
          >
            城市
          </label>
          <input
            className="form-control"
            id="searchCity"
            onChange={[Function]}
            placeholder="城市"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </form>
  <div
    className="container"
  >
    <div
      className="container-list"
    >
      <span
        className="list-summary"
      >
        找到0份和您所填经历匹配的工作
      </span>
      <ul
        className="pagination"
      />
    </div>
    <div
      className="container-details"
    />
  </div>
</div>
`;
